### DISCUSSION TITLE
Add support for the new Smoke+CO Detector

### DISCUSSION THREAD

**Discussion by Abductist** [homebridge-simplisafe3/homebridge-simplisafe3#390] [Feature Requests]
### Description

The new Smoke+CO detector (type 14) should be added to simplipy.device.DeviceTypes. In addition to triggering the same fire (1110) and CO alarm (1162) events that the individual smoke and CO detectors can trigger, this device can also trigger a "pre-alarm" event (1901 with zoneCid == 1 meaning it was triggered and zoneCid == 0 meaning it was cleared) although I'm not sure how beneficial adding that would be.

### Expected behavior

The new Smoke+CO detector should work properly in Homekit

### Steps to reproduce

Pair Smoke+CO device to system and notice the "Sensor not (yet) supported" error message in the log

### Homebridge logs

```bash
{
"flags": {
"offline": true,
"lowBattery": false,
"swingerShutdown": false
},
"serial": "00000000",
"type": 14,
"name": "Kitchen",
"setting": {},
"status": {
"coTriggered": false,
"malfunction": false,
"tamper": true,
"lowSensitivity": false,
"endOfLife": false,
"test": false,
"smokeTriggered": false,
"preSmokeAlarm": false
},
"timestamp": 0,
"rssi": -45,
"WDTCount": 23,
"nonce": 0,
"rebootCnt": 17,
"deviceGroupID": 0
}
```

### Homebridge config.json contents

```json
{
    "name": "System",
    "cameras": true,
    "sensorRefresh": 60,
    "persistAccessories": true,
    "cameraOptions": {
        "enableHwaccelRpi": false
    },
    "excludedDevices": [],
    "platform": "homebridge-simplisafe3.SimpliSafe 3"
}
```

### Plugin version

1.10.13

### Homebridge version

1.6.0

### Node.js version

18.16.0

### Host OS

Windows

**Comment by snoslicer8** [comment]
I would also love to request this integration. I just upgraded one detector on each floor of my home to the new Smoke+CO units and lost them out of my Homebridge configuration. I know this takes time, so happy to wait, but expressing my interest!

**Comment by snoslicer8** [comment]
Follow-up on this request. Is CO+Smoke detector a sensor that can be added to this plugin, please?

**Comment by asabatino88** [comment]
I too would like this feature added, but I'm not sure it's necessary. What would the benefit be? Right now, if a Simplisafe Smoke + CO gets triggered by the Simplisafe system, the alarm goes off and the plugin alerts via HomeKit. So for that purpose, the base functionality works. What would adding the individual smoke+co devices to HomeKit accomplish? Perhaps there are sensors in there that could be exposed (maybe a temperature or air quality sensor), but otherwise, I'm not sure the benefit. The only thing that would be nice for my use case is cleaning up the Homebridge logs at restart, as I get 8 "device currently not supported" entries.

**Comment by seeloesix** [comment]
It would be great to have this one added.  Anything we can provide or help with?

**Comment by evantahler** [comment]
Hey! Are you looking for help to add these?  

```
[3/22/2026, 10:32:04 AM] [Home Alarm] Sensor not (yet) supported: Basement
[3/22/2026, 10:32:04 AM] [Home Alarm] {
  flags: { offline: false, lowBattery: false },
  serial: '02bcee1e',
  type: 14,
  name: 'Basement',
  setting: {},
  status: {
    coTriggered: false,
    malfunction: false,
    tamper: false,
    lowSensitivity: false,
    endOfLife: false,
    test: false,
    smokeTriggered: false,
    preSmokeAlarm: false
  },
  deviceGroupID: 0
}
```

**Reply by shamoon** [reply]
I don’t have any of them to test with, I assume it would pretty straightforward to do if so. If you have dev experience we’re always open to PRs
